pkg/apiroutes/templates.go

```diff
@@ -355,3 +357,23 @@ func BatchPatchTemplateRepos(conID string, operations []RepoOperation) ([]SubRes
 
 	return subResponsesFromBatchOperation, nil
 }
+
+// HTTPRequestWithRetryOnLock : Retries a request until either the maxHTTPRetries limit has been hit or the response status code is not equal to 423 (templates locked HTTP code)
+func HTTPRequestWithRetryOnLock(httpClient utils.HTTPClient, originalRequest *http.Request, connection *connections.Connection) (*http.Response, *sechttp.HTTPSecError) {
+	// maxHTTPRequests is the total number of requests, 1 for the intial and the rest are retries
+	var maxHTTPRequests = 5
+	var templatesLockedHTTPStatusCode = http.StatusLocked
+	var response *http.Response
+	for i := 0; i < maxHTTPRequests; i++ {
+		resp, httpSecError := sechttp.DispatchHTTPRequest(httpClient, originalRequest, connection)
+		if httpSecError != nil {
+			return nil, httpSecError
+		} else if resp.StatusCode != templatesLockedHTTPStatusCode || i+1 == maxHTTPRequests {
+			response = resp
+			break
+		} else {
+			time.Sleep(time.Second)
+		}
+	}
+	return response, nil
```

Comment:
will it ever hit this?

Reply:
can see why it's in though, as you need a response at the end of the function

Reply:
yes its the only return that returns the response